### PR TITLE
python312Packages.blessings: drop

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description of changes

- #326513

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by emilazy** [comment]
This package seems to be somewhat famously unmaintained, and there is a maintained fork in the form of [Blessed](https://github.com/jquast/blessed) that already uses pytest. We already have a `python3Packages.blessed` package; perhaps we can simply remove this. In‐tree users:

* `cbeams`: Doesn’t look very maintained, but [`terminal.py`](https://github.com/tartley/cbeams/blob/516d1f708c7d187b9a32c40e96745d31bfe3b472/cbeams/terminal.py) seems simple enough that I expect the port is simple.
* `python3Packages.invocations`: https://github.com/pyinvoke/invocations/issues/38, looks like probably a one‐line import change to port to Blessed.
* `python3Packages.ronin`: Build system with no commits for 7 years and no in‐tree users, not exposed in the top‐level package set, ditto on the port but we can probably just drop this.
* `python3Packages.reportengine`: Marked as broken, unmaintained‐ish upstream, and could potentially just be dropped, but ditto on the port.

**Comment by Sigmanificient** [comment]
@emilazy tried to makes `reportengine` passing, couldn't successfully patch it, so I dropped it. It also takes 2 other packages with it

**Comment by Sigmanificient** [comment]
Result of `nixpkgs-review pr 336152` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>cbeams</li>
    <li>cbeams.dist</li>
    <li>python311Packages.invocations</li>
    <li>python311Packages.invocations.dist</li>
    <li>python311Packages.pytest-relaxed</li>
    <li>python311Packages.pytest-relaxed.dist</li>
    <li>python312Packages.invocations</li>
    <li>python312Packages.invocations.dist</li>
    <li>python312Packages.pytest-relaxed</li>
    <li>python312Packages.pytest-relaxed.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/blessings/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/blessings/default.nix:1]: Please fix the commit message

File: pkgs/top-level/python-aliases.nix [hunk 1]
```diff
@@ -513,6 +513,7 @@ mapAliases ({
   recursivePthLoader = recursive-pth-loader; # added 2024-01-07
   rednose = throw "rednose is no longer maintained (since February 2018)"; # added 2023-08-06
   repeated_test = repeated-test; # added 2022-11-15
+  reportengine = throw "reportengine has been removed, since it is unmaintained and broken"; # added 2024-08-21
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-aliases.nix:517]: Please name it explicitly

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -1522,6 +1522,7 @@ mapAliases ({
   ### V ###
 
   v4l_utils = throw "'v4l_utils' has been renamed to/replaced by 'v4l-utils'"; # Converted to throw 2023-09-10
+  validphys2 = throw "validphys2 has been removed, since it has a broken dependency that was removed"; # Added 2024-08-21
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:1525]: same here

File: pkgs/top-level/python-aliases.nix [hunk 3]
```diff
@@ -327,6 +327,7 @@ mapAliases ({
   mrkd = throw "mrkd has been promoted to a top-level attribute name: `pkgs.mrkd`"; # added 2023-08-01
   multi_key_dict = multi-key-dict; # added 2023-11-05
   mutmut = throw "mutmut has been promoted to a top-level attribute name: `pkgs.mutmut`"; # added 2022-10-02
+  n3fit = throw "n3fit has been removed since it relies on a dependency that was removed"; # added 2024-08-24
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-aliases.nix:330]: and here

File: pkgs/top-level/python-aliases.nix (hunk 1 above)
Comment (emilazy) [reply pkgs/top-level/python-aliases.nix:517]: What do you mean?

Comment (Sigmanificient) [reply pkgs/top-level/python-aliases.nix:517]: I am not sure to see what is problematic here

**Review by emilazy — APPROVED** [review]
Could you fix the PR title? Otherwise this seems good to merge to me.

**Comment by emilazy** [comment]
Result of `nixpkgs-review pr 336152` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>cbeams</li>
    <li>cbeams.dist</li>
    <li>python311Packages.invocations</li>
    <li>python311Packages.invocations.dist</li>
    <li>python311Packages.pytest-relaxed</li>
    <li>python311Packages.pytest-relaxed.dist</li>
    <li>python312Packages.invocations</li>
    <li>python312Packages.invocations.dist</li>
    <li>python312Packages.pytest-relaxed</li>
    <li>python312Packages.pytest-relaxed.dist</li>
  </ul>
</details>

**Review by emilazy — APPROVED** [review]
Remembered I can do that myself :)

**Pull request merged**